develop-docs/VIEW_MASKING_STRATEGIES.md

```diff
@@ -0,0 +1,1150 @@
+- No dependence on globally toggled system services (e.g., VoiceOver) in production.
+- Work reliably in background-capable scenarios and during animations (e.g., during view transitions).
+- Capture edge-cases such as frames taken during view transitions.
+- Be configurable via user-provided include/exclude class lists.
+
+## Processing Pipeline Overview
+
+The process when taking a screenshot or session replay frame is as follows:
+
+```mermaid
+flowchart TD
+    A[**Root View/Window**]
+    B[**Full-Screen Screenshot**<br/>UIImage from root view/window]
+    C[**Region Detection Engine**<br/>Accessibility, View Hierarchy, ML]
+    D[**Mask Renderer**<br/>Solid overlay of redact regions]
+    E[**Masked Output Image**]
+
+    A --> B --> C --> D --> E
+```
+
+This document focuses on the **Region Detection Engine** and how we calculate the `SentryRedactRegion` geometry based on a given view hierarchy and layer tree.
+
+## View Hierarchies on iOS
+
+On iOS, the UI is organized as a tree of `UIView` objects (UIKit views). Each view manages layout, interaction, and is part of the responder chain. Every `UIView` also owns a primary Core Animation layer (`CALayer`) available via `view.layer`. Layers form their own tree and are responsible for rendering and animation.
+
+Difference between a view and a layer:
+
+- View (`UIView`): UIKit object that participates in hit‑testing and input events, Auto Layout, coordinate conversion, and accessibility. Views determine where content goes and when it needs to be redrawn, and may implement `draw(_:)` or act as the `CALayer`’s delegate.
+- Layer (`CALayer`): Core Animation object that holds the visual content and animation state. Layers composite on the render server, expose visual properties (e.g., `backgroundColor`, `cornerRadius`, `shadow*`, `opacity`, `transform`), and can have `sublayers`. Layers do not receive touch events, are not first responders, and do not own constraints; their geometry is defined by `bounds`/`position`/`anchorPoint`.
+
+Important consequences:
+
+- The view tree and the layer tree are related but not identical. A single view can add multiple sublayers, and some sublayers have no backing `UIView`.
+- During animations, the on‑screen geometry lives in the layer’s `presentationLayer` (the animated state), which can differ from the model layer values and the view’s frame.
+- Frameworks like SwiftUI and hybrid renderers often draw into generic views that host complex layer subtrees without exposing a view hierarchy.
+
+These distinctions matter for masking: to match what the user actually sees, our redaction logic inspects the Core Animation layer tree (and, when needed, the `presentationLayer`) rather than relying only on the `UIView` hierarchy.
+
+## View Hierarchy & Layer Tree-Based Redaction
+
+> ![INFO]
+> As of Nov 13th, 2025, this is the default implementation for identifying which areas of a view hierarchy should be masked during screenshot or session replay capture.
```

Comment:
`l`: I'm unsure if that's the formatting you want to have
<img width="2004" height="262" alt="Image" src="https://github.com/user-attachments/assets/4310e42e-d9ee-4762-8e2c-9b43859d5be6" />